src/api/schema/metrics/filter.rs

```diff
@@ -68,6 +71,15 @@ impl<'a> MetricsFilter<'a> for Vec<Metric> {
         Some(EventsInTotal::new(sum))
     }
 
+    fn received_events_total(&self) -> Option<ReceivedEventsTotal> {
+        let sum = sum_metrics(
+            self.iter()
+                .filter(|m| m.name() == "component_received_events_total"),
```

Comment:
Good, with the name of the PR I had to check if you saw the new component metric names.

Reply:
Yeah, it's a little ambiguous -- `received_events_total` here refers to the API field name, rather than the underlying metric, since (in most cases) it's already nested under a request for the component:

```
subscription ComponentReceivedEventsTotalsSubscription($interval: Int!) {
    componentReceivedEventsTotals(interval: $interval) {
        pipelineId
        componentId
        metric {
            receivedEventsTotal
        }
    }
}
```